packages/ui/libs/ui/core/src/components/icons/DeviceIcon.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { ReactComponent as IconWind } from '../../../assets/device/icon-wind.svg';
+import { ReactComponent as IconHydro } from '../../../assets/device/icon-hydro.svg';
+import { ReactComponent as IconThermal } from '../../../assets/device/icon-thermal.svg';
+import { ReactComponent as IconSolid } from '../../../assets/device/icon-solid.svg';
+import { ReactComponent as IconLiquid } from '../../../assets/device/icon-liquid.svg';
+import { ReactComponent as IconGaseous } from '../../../assets/device/icon-gaseous.svg';
+import { ReactComponent as IconMarine } from '../../../assets/device/icon-marine.svg';
+import { ReactComponent as IconSolar } from '../../../assets/device/icon-solar.svg';
+import { StringMap } from 'i18next';
+
+const ElementsDictionary: Record<string, JSX.Element> = {
+  Wind: <IconWind />,
+  'Hydro-electric Head': <IconHydro />,
+  Thermal: <IconThermal />,
+  Solid: <IconSolid />,
+  Liquid: <IconLiquid />,
+  Gaseous: <IconGaseous />,
+  Marine: <IconMarine />,
+};
+
+const tooltipDictionary: Record<string, [string, StringMap]> = {
```

Comment:
This is not a core component, because icons and types are specific to the type of registry and backend.